File: Minimum Number of Browsers/minimum_num_browsers.cs
/*
    Minimum number of browsers, C#
    Module written by Brandon Arvanaghi 
    Website: arvanaghi.com 
    Twitter: @arvanaghi
*/

using System;
using Microsoft.Win32;

namespace MinimumNumBrowsersChecker {
    class Program {
        static void Main(string[] args) {
            int browserCount = 0;
            string[] browserKeys = {@"SOFTWARE\Microsoft\Windows\CurrentVersion\App Paths\chrome.exe", @"SOFTWARE\Microsoft\Windows\CurrentVersion\App Paths\iexplore.exe", @"SOFTWARE\Mozilla"};

            foreach (string browserKey in browserKeys) {
                RegistryKey OpenedKey = Registry.LocalMachine.OpenSubKey(browserKey, false);
                if (OpenedKey != null) {
                    browserCount += 1;
                }
            }

            if (browserCount >= 2){
                Console.WriteLine("Proceed!");
                Console.ReadKey();
            } else {
                Console.WriteLine("Number of Browsers: {0}", browserCount);
                Console.ReadKey();
            }
        }
    }
}

File: Minimum Number of Browsers/minimum_num_browsers.rb
#
#   Minimum number of browsers, Ruby
#   Module written by Brandon Arvanaghi
#   Website: arvanaghi.com 
#   Twitter: @arvanaghi
#

require 'win32/registry'

browserCount = 0

browserKeys = ['SOFTWARE\Microsoft\Windows\CurrentVersion\App Paths\chrome.exe', 'SOFTWARE\Microsoft\Windows\CurrentVersion\App Paths\iexplore.exe', 'SOFTWARE\Mozilla']

for index in 0 ... browserKeys.size
  begin
    reg_access = Win32::Registry::KEY_READ | 0x100
    Win32::Registry::HKEY_LOCAL_MACHINE.open(browserKeys[index], reg_access) { browserCount += 1 }
  rescue
  end
end

if browserCount >= 2
  puts "Proceed!"
else
  puts "Number of Browsers: " + browserCount.to_s
end